src/AnalyticsContext.tsx

```diff
@@ -35,18 +47,39 @@ const AnalyticsProvider: FunctionComponent = ({ children }) => {
     checkAnalyticsConsent()
   }, [userConsentedToAnalytics])
 
+  useEffect(() => {
+    supportAnalyticsTracking &&
+      Matomo.initTracker(
+        healthAuthorityAnalyticsUrl,
+        healthAuthorityAnalyticsSiteId,
+      )
+  }, [
+    supportAnalyticsTracking,
+    healthAuthorityAnalyticsSiteId,
+    healthAuthorityAnalyticsUrl,
+  ])
+
   const updateUserConsent = async (consent: boolean) => {
     await StorageUtils.setAnalyticsConsent(consent)
     setUserConsentedToAnalytics(consent)
   }
 
   const trackEvent = async (event: string) => {
-    return userConsentedToAnalytics && actions.trackEvent(event)
+    return supportAnalyticsTracking && actions.trackEvent(event)
+  }
+
+  const trackScreenView = async (screen: string) => {
+    supportAnalyticsTracking && actions.trackScreenView(screen)
```

Comment:
How do you feel about creating a `voidState` of sorts and then on the initial mount we provide that value to the provider unless analytics tracking is supported? Does it feel like having the short circuit in both these places might make it hard to follow? Or is it better to have the checks right in front of the tracking event?